src/Database/Redis/ProtocolPipelining.hs

```diff
@@ -93,15 +96,20 @@ connect hostName portID timeoutOpt =
                 Right () -> do
                   phase <- readMVar phaseMVar
                   errConnectTimeout phase
-        hConnect' (PortNumber port) mvar =
+        hConnect' port mvar =
           bracketOnError mkSocket NS.close $ \sock -> do
             NS.setSocketOption sock NS.KeepAlive 1
             void $ swapMVar mvar PhaseResolve
-            host <- BSD.getHostByName hostName
+            hostAddrInfos <- NS.getAddrInfo Nothing (Just hostName) Nothing
+            let getAddr = \ case
+                  [] -> throwIO $ ConnectHostNotFound hostName
+                  x:xs -> case NS.addrAddress x of
+                    NS.SockAddrInet _ addr -> pure addr
+                    _ -> getAddr xs
+            hostAddr <- getAddr hostAddrInfos
```

Comment:
Don't know if this is good or bad or if it has implicit assumptions that are wrong. Wasn't able to find any guidance from the network package on upgrading.

Reply:
Oh well. I guess I'll have to give this a thoughtful review, and maybe it's worth asking the network package maintainers why was the drastic change not announced properly.